proposals/threads/Overview.md

```diff
@@ -48,8 +48,48 @@ agent's cluster will have access to the additional linear memory.
 When a module has an imported linear memory, its data segments are copied into
 the linear memory when the module is instantiated.
 
-When the imported linear memory is shared, the writes are non-atomic and
-are not ordered.
+When linear memory is not shared, the ordering of data segment initialization is
+not observable. When linear memory is shared, it is possible for another module
```

Comment:
That's not true: you can observe an OOB segment in the corresponding memory (or rather, observe all the segments which were initialized before it OOB'd). What's true is that you can't observe order within a segment.

I think what you specify below could just apply to non-shared as well.